### ISSUE TITLE
Solver returns 0.0S/s after pool reconnects

### ISSUE THREAD

**Issue by RodrigoSerafim** [Genoil/ZECMiner#40]
When a pool forcibly disconnects the client, the miner begins reconnecting to the pool.
However, when the connection is re-established the solver stops working and outputs 0.0S/s for all GPU cores.
Killing the process and restarting it resolves this issue and the miner immediately resumes solving eqhash, suggesting there is a bug in the re-connection logic.
I have attached the console output of an execution run to better find a clue on the origin of the bug (wallet addr redacted).
[console_output.zip](https://github.com/Genoil/ZECMiner/files/560869/console_output.zip)

**Comment by masoodarny** [comment]
i have the same problem but no solution from anyone//////

**Comment by RodrigoSerafim** [comment]
As a workaround, while Genoil tracks the bug, I have writen a Windows Poweshell script to monitor the miner output. Instead of restarting every 15 minutes it only restarts on repeated appearances of the 0.0S/s string.
I have tested it with a ping command and it works but haven't seen an error in genoil.exe since.
I need to go to sleep for now, but if anyone can verify it works I appreciate it.
[monitor_restart.zip](https://github.com/Genoil/ZECMiner/files/560987/monitor_restart.zip)

**Comment by RockZhou8** [comment]
i've download `monitor_restart.zip` and unzip the `.ps1` file
i run in the powershell and got 15 counts 0.0S/s
but the script stop at red line `--KILLING PROCESS AND RESTARTING--` ......
no restart at all...
would you plz check the script and fix it ?thx bro!

**Comment by RodrigoSerafim** [comment]
Yep, got the chance to test the script a little better and I think this version works. Powershell has some weird way to deal with variable scopes it seems. At least it held for almost 16 hours and I saw it recover the process after a row of 0.0S/s.
I have attached a new version of the script.
Feel free to alter the search condition to any other string that suits your specific problem better.
[monitor_restart_v2.zip](https://github.com/Genoil/ZECMiner/files/562711/monitor_restart_v2.zip)

**Comment by RockZhou8** [comment]
thx!i'll test the v2 version

**Comment by siteuper** [comment]
If you remove "total" from text search string, script will be work in the case then only one card give 0.0S/s, or two, or any.
